build.ps1

```diff
@@ -109,16 +109,24 @@ $gitCommit = git rev-parse HEAD
 git archive --format=zip -o "$OUTPUT_DIR\\dotnet-packages-sources.zip" $gitCommit
 "dotnet-packages.git $gitCommit" | Out-File -FilePath "$OUTPUT_DIR\dotnet-packages-manifest.txt"
 
+# prepare NuGet Config
+
+$RESTORE_NUGET_CONFIG_FILE="NuGet.Config"
+
+((Get-Content -path $RESTORE_NUGET_CONFIG_FILE -Raw) -replace 'https://api.nuget.org/v3/index.json', $NugetSource) |`
+ Set-Content -Path $RESTORE_NUGET_CONFIG_FILE
+
 #prepare sharpziplib
 
-mkdirClean "$SCRATCH_DIR\sharpziplib"
-Expand-Archive -DestinationPath "$SCRATCH_DIR\sharpziplib" -Path "$REPO\SharpZipLib\SharpZipLib-1.3.3.zip"
+mkdirClean "$SCRATCH_DIR\sharpziplib"	
+Expand-Archive -DestinationPath "$SCRATCH_DIR\sharpziplib" -Path "$REPO\SharpZipLib\SharpZipLib-1.3.3.zip"	
 
-Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-sharpziplib") } | % { $_.FullName } |`
-  applyPatch -Path "$SCRATCH_DIR\sharpziplib\SharpZipLib-1.3.3"
+Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-sharpziplib") } | % { $_.FullName } |`	
+  applyPatch -Path "$SCRATCH_DIR\sharpziplib"	
 
-msbuild $SWITCHES $FRAME48 $VS_TOOLS $SIGN "$SCRATCH_DIR\sharpziplib\SharpZipLib-1.3.3\ICSharpCode.SharpZipLib.sln"
-'dll', 'pdb' | % { "$SCRATCH_DIR\sharpziplib\bin\ICSharpCode.SharpZipLib." + $_ } |`
+dotnet restore -v n "$SCRATCH_DIR\sharpziplib\SharpZipLib-1.3.3\src\ICSharpCode.SharpZipLib\ICSharpCode.SharpZipLib.csproj"
```

Comment:
since you're passing `$RESTORE_SWITCHES -p:RestoreConfigFile=$RESTORE_NUGET_CONFIG_FILE`, you don't need `dotnet restore`